### PR TITLE
CODENVY-31: Replaced messages from git importer on error codes

### PR DESCRIPTION
**Pull request by akorneta**
Also added git checkout event which goes through by websocket connection

@skabashnyuk @vparfonov

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok for me. @vparfonov @sleshchenko ?

File: core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitWebSocketMessenger.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * @author Anton Korneta.
+ */
+@Singleton
+public class GitWebSocketMessenger implements EventSubscriber<GitCheckoutEvent> {
+    private static final Logger LOG = LoggerFactory.getLogger(GitWebSocketMessenger.class);
+    private static final String CHANNEL = "git:events:channel";
```
Comment (sleshchenko) [inline core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitWebSocketMessenger.java]: You should use different channels for each project in each workspace

File: core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java [hunk 2]
```diff
@@ -323,11 +325,20 @@ private void sparsecheckout(GitConnection git,
...
         new File(project, ".gitignore").delete();
     }
+
+    private String getRemoteBranch(DtoFactory dtoFactory, GitConnection git, String branchName) throws GitException {
+        final List<Branch> remotes = git.branchList(dtoFactory.createDto(BranchListRequest.class)
+                                                              .withListMode(BranchListRequest.LIST_REMOTE));
+        final Optional<Branch> first = remotes.stream()
+                                              .filter(br -> branchName.equals(br.getDisplayName()))
+                                              .findFirst();
+        return first.get().getName();
```
Comment (sleshchenko) [inline core/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java:350]: Think about NoSuchElementException here. Is it possible here?

**Comment by sleshchenko** [comment]
other OK

**Comment by skabashnyuk** [comment]
ок

**Pull request merged**